src/test/tests.ts

```diff
@@ -213,6 +216,40 @@ describe('Client', function () {
     new SubscriptionClient(`ws://localhost:${RAW_TEST_PORT}/`);
   });
 
+  it('should send INIT message first, then the SUBSCRIPTION_START message', (done) => {
+    let initReceived = false
+
+    wsServer.on('connection', (connection: any) => {
+      connection.on('message', (message: any) => {
+        const parsedMessage = JSON.parse(message);
+        // mock server
+        if (parsedMessage.type === INIT) {
+          connection.send(JSON.stringify({type: INIT_SUCCESS, payload: {}}));
+          initReceived = true
+        }
+        if (parsedMessage.type === SUBSCRIPTION_START) {
+          expect(initReceived).to.be.true
+          done()
```

Comment:
Need to setup the client as a function variable and include a `client.unsubscribeAll();` prior to calling `done()` otherwise the client will stick around and try to reconnect.